modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/DeploymentDirectoryWatcher.java

```diff
@@ -135,6 +137,16 @@ public DeploymentDirectoryWatcher(DomainBundleArchiveDeployer domainBundleDeploy
     this.schedulerServiceSupplier = schedulerServiceSupplier;
     this.applicationTimestampListener = new ArtifactTimestampListener(applications);
     this.domainTimestampListener = new ArtifactTimestampListener(domains);
+    this.artifactStoppedDeploymentListeners = new ArrayList<>();
+  }
+
+  private static int getChangesCheckIntervalMs() {
+    try {
+      String value = System.getProperty(CHANGE_CHECK_INTERVAL_PROPERTY);
+      return Integer.parseInt(value);
```

Comment:
Use `Integer.getInteger` to read the system property.

Reply:
Tried it but `Integer.getInteger` returns null and a NullPointerException is thrown 🤔